pkg/recipes/terraform/module.go

```diff
@@ -53,12 +50,18 @@ type moduleInspectResult struct {
 // localModuleName is the name of the module specified in the configuration used to download the module.
 // It uses terraform-config-inspect to load the module from the directory. An error is returned if the module
 // could not be loaded.
-func inspectModule(workingDir, localModuleName string) (*moduleInspectResult, error) {
+func inspectModule(workingDir, localModuleName string, templatePath string) (*moduleInspectResult, error) {
+	moduleRootDir := ".terraform/modules"
 	result := &moduleInspectResult{ContextVarExists: false, RequiredProviders: []string{}, ResultOutputExists: false, Parameters: map[string]any{}}
 
 	// Modules are downloaded in a subdirectory in the working directory.
 	// Name of the module specified in the configuration is used as subdirectory name.
 	// https://developer.hashicorp.com/terraform/tutorials/modules/module-use#understand-how-modules-work
+	//
+	// If the template path is for a submodule, we'll add the submodule path to the module root directory.
+	if strings.Contains(localModuleName, "//") {
```

Comment:
Terraform has a hard convention for the double slash: https://developer.hashicorp.com/terraform/language/modules/sources#modules-in-package-sub-directories

We can use this special character to check if the module source is a submodule

Reply:
Updated this to find the last index of `//` but the logic for using those characters as a delineator is the same